File: CPP/Sorting/selectionSort.cpp
#include<iostream>
#include<stdio.h>

using namespace std;

void printArray(int arr[], int len){
    for (int i = 0; i<len; i++){
        cout<<arr[i]<<" ";
    }
    return;
}

void swap(int *a, int *b){
            int temp;
            temp = *a;
            *a = *b;
            *b = temp;
        }

void SelectionSort(int arr[], int len){
    
    for(int i = 0; i < len-1; i++){

        int minindex = i;
        for(int j = i+1; j < len; j++){

            if(arr[j] < arr[minindex]){
                minindex = j;
            }

            if(minindex != i){
                swap(&arr[minindex], &arr[i]);
            }
        }
    }
    return ;
}

int main()
{
    int arr[] = {4, 3, 5, 13, 0, 11, 2, 7, 8, 9, 6, 1};
    // int arr[] = {4, 3, 5, 9, 6, 1};

    int len = sizeof(arr)/sizeof(arr[0]);
    cout<<len<<endl;
    printArray(arr, len);
    cout<<endl;
    SelectionSort(arr, len);
    printArray(arr, len);
    return 0;
}

File: OLD_DSA/PYTHON/sortAlgo/selectionSort.py
rawArray = [5, 3, 7, 1, 0, 19, -1]


def selectionSort(arr):
    
    for i in range(len(arr)):
        print(rawArray)
        min_index = i
        for j in range(i+1, len(arr)):
            if arr[min_index] > arr[j]:
                min_index = j
        arr[min_index], arr[i] = arr[i], arr[min_index]
    return arr

# selectionSort(rawArray)